src/libs/masks.c

```diff
@@ -737,16 +737,58 @@ static void _tree_moveup(GtkButton *button, dt_lib_module_t *self)
   GtkTreeModel *model = gtk_tree_view_get_model(GTK_TREE_VIEW(lm->treeview));
   GtkTreeSelection *selection = gtk_tree_view_get_selection(GTK_TREE_VIEW(lm->treeview));
   GList *items = gtk_tree_selection_get_selected_rows(selection, NULL);
-  for(const GList *items_iter = items; items_iter; items_iter = g_list_next(items_iter))
+
+  for(const GList *items_iter = items;
+      items_iter;
+      items_iter = g_list_next(items_iter))
   {
     GtkTreePath *item = (GtkTreePath *)items_iter->data;
     GtkTreeIter iter;
+    GtkTreeIter prev_iter;
     if(gtk_tree_model_get_iter(model, &iter, item))
     {
       int grid = -1;
       int id = -1;
       _lib_masks_get_values(model, &iter, NULL, &grid, &id);
 
+      prev_iter = *gtk_tree_iter_copy(&iter);
```

Comment:
Looks like a helper function could avoid some duplication with almost same in `_tree_movedown` below?

Reply:
Yes, you're certainly right. I should have set this to draft anyway as I also need to free iter/path memory.